### PR TITLE
jimmy/fix/fix-camping-section-in-rec-resource-page

### LINKED ISSUE #552: Improve behavior for camping section on resource page
Improvements for camping section on resource page
Based on designs from #283 

**Acceptance Criteria**
- [x] When camping shows up in activities but there are no fees and no number of campsites, do not display the 'camping' header/section
- [x] When there are no camping fees but there are campsites, remove the 'camping fees' section as seen in the mockups: https://www.figma.com/design/TFDtavdpJE2vrB5vEmBB6w/RST---Basic-View?node-id=749-14057&t=1A7LGY4TujAD3dPE-4
- [x] Related to above, be sure to fix bug here:
     - [x] Camping section shows on resource page when there is no camping/fees/#of campsites (no camping data at all for REC#). Should it show at all?
0 k snowmobile parking lot
Recreation Site | REC204117
nahwitti river trail
Recreation Site | REC0149

Image

**Developer Checklist**
- [ ] refactor additional fees and don't duplicate camping component

### PR DESCRIPTION
**Pull request by jimmypalelil**
# Description

Updated the rendering logic to only show camping and additional fees sections when required data is present.

## Changes
- Camping section displayed when either:
    - Recreation fees exist
    - Campsite count > 0
- Additional Fees section displayed when:
    - Additional fees array contains items

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)

## Testing
Added comprehensive test cases covering:
- Camping section rendering with various combinations of fees and campsite counts
- Additional fees section rendering with empty/populated fee arrays
- Edge cases for both sections

## Further Comments

Added Storybook integration to improve component development and testing:

- Created stories for both Camping, Additional Fees, and RecResourcePage components

### CODE REVIEW

**Review by marcellmueller — CHANGES_REQUESTED** [review]
Storybook seems out of scope for this. Definitely not opposed to using it but don't you think it's a decision that should be brought up with the team and product owner?  
  
Can you make a storybook ticket and PR and lead a discussion about the benefits of it and get buy in?  
- What are the benefits of story book
- Team agreements on what and when to test
- How will we keep this up to date? I've used it on projects where we basically forgot about it and it got broken as we didn't keep it up to date with component changes. Github actions?

**Review by marcellmueller — APPROVED** [review]
This is working well, thanks for the refactor 🙇

File: README.md [hunk 1]
```diff
@@ -246,3 +247,32 @@ Schedule job runs every saturday and keep schema documentation upto date in
 ## Architecture
 
 ![RST Architecture](docs/rst-arch.drawio.svg)
+
+## Storybook Integration
```
Comment (jimmypalelil) [inline README.md:251]: Stories do not need to be updated or maintained. They are simply development tools for testing different states when browser-based API overrides are not enough.
